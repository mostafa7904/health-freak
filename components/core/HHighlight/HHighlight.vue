<template>
  <mark class="h-highlight">
    <slot> </slot>
  </mark>
</template>

<script lang="ts">
import { defineComponent } from "vue";

export default defineComponent({
  name: "HealthFreakHighlight",
});
</script>

<style lang="scss">
.h-highlight {
  background-image: url("/svg/highlight.svg");
  background-color: transparent;
  background-position: bottom;
  background-repeat: no-repeat;
  color: inherit;
}
</style>
