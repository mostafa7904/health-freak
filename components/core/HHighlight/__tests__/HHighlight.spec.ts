// Libraries
import Vue from "vue";

// Components
import HHighlight from "../HHighlight.vue";

// Utilities
import { createLocalVue, mount, Wrapper } from "@vue/test-utils";

describe("HHighlight.vue", () => {
  let mountFunction: (options?: object) => Wrapper<Vue>;

  let localVue: typeof Vue;

  beforeEach(() => {
    localVue = createLocalVue();

    mountFunction = (options = {}) => {
      return mount(HHighlight, {
        localVue,
        ...options,
      });
    };
  });

  it("should render component and match snapshot", () => {
    const html = mountFunction().html();
    expect(html).toMatchSnapshot();
  });
});
