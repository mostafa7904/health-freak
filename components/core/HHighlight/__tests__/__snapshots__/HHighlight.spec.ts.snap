// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HHighlight.vue should render component and match snapshot 1`] = `"<mark class=\\"h-highlight\\"></mark>"`;
